apps/ex_nvr_web/lib/ex_nvr_web/live/device_list_live.ex

```diff
@@ -99,11 +103,21 @@ defmodule ExNVRWeb.DeviceListLive do
   end
 
   def handle_event("stop-recording", %{"device" => device_id}, socket) do
-    update_device_state(socket, device_id, :stopped)
+    user = socket.assigns.current_user
+
+    case authorized?(user, :device, :update) do
+      {:ok, :authorized} -> update_device_state(socket, device_id, :stopped)
+      {:error, :unauthorized} -> unauthorized(socket, ~p"/devices", :noreply)
```

Comment:
I don't think we need redirect, a flash message is sufficient (also hide this option on the live view of course).

Reply:
You still redirect ???